scenario/mocking.py

```diff
@@ -73,26 +73,40 @@ class ActionMissingFromContextError(Exception):
 
 
 class _MockExecProcess:
-    def __init__(self, command: Tuple[str, ...], change_id: int, out: "ExecOutput"):
+    def __init__(
+        self,
+        command: Tuple[str, ...],
+        change_id: int,
+        exec: "Exec",
```

Comment:
we're shadowing the exec builtin

Reply:
True, but only within this one small function, which seems very unlikely to ever want to use the built-in exec. @benhoyt and I have talked about this before - if I recall correctly he's more relaxed about shadowing than I am and would do it anywhere it's not going to cause problems or confusion, while I'd prefer to avoid it in general. Here it seems like it's the clearest name for the argument (rather than `exc` for example).

It does bring up the more general case where people are using this, or we're providing examples, like:

```python
exec = Exec(["sleep", "10"])
container = Container(execs={exec})
state = ...
```

My guess is that Ben's totally fine with that in a test, whereas I would cringe a little. Maybe for docs it would be better to encourage this?

```python
sleep = Exec(["sleep", "10"])
any_exec = Exec([], stdout=...)
```

I would still prefer to have the full name "exec" in `_MockExecProcess`'s argument list, though.

Reply:
Yeah, I'm more relaxed about this, especially when it's about parameter names -- I'd much rather have a good parameter name (especially for keyword arguments) and shadow a builtin than have a poorer parameter name just for the sake of not shadowing.

In this particular case the builtin we're shadowing is `exec()`, which is *extremely* rare, so I'd definitely prioritise having a good param name here. Where it's something common like `min` or `len` or `all` I'm more on the "we probably shouldn't shadow those" side.

I think the case for shadowing is weaker when it's just a local variable (and not a parameter). There it's easy to rename internally.